nntrainer/tensor/tensor.cpp

```diff
@@ -633,9 +633,33 @@ Tensor &Tensor::dot(Tensor const &m, Tensor &result, bool trans,
   float *rdata = result.getData();
   const float alpha = 1.0f;
   const float beta = 0.0f;
-
   enum CBLAS_TRANSPOSE transA = trans ? CblasTrans : CblasNoTrans;
   enum CBLAS_TRANSPOSE transB = trans_m ? CblasTrans : CblasNoTrans;
+
+  /// shortcut calculation in case of vector
+  /// case1: (1 * K) X (K * 1)
+  if (M == 1 && N == 1) {
+    *rdata = sdot(K, data, 1, mdata, 1);
+    return result;
+  }
+
+  /// case2: (M * K) X (K * 1)
+  if (N == 1) {
+    sgemv(CblasRowMajor, transA, dim1, dim2, alpha, data, lda, mdata, 1, beta,
+          rdata, 1);
+    return result;
+  }
+
+  /// case3: (1 * K) X (K * N) = 1 * N = R
+  /// = R^T = (K * N) ^T * (1 * K)
```

Comment:
this will become `(K * N) ^T * (1 * K) ^ T` -> `N * K X K * 1` right? S
Just like above, please add check for transA or add comment that its not needed.

Reply:
I'll add additional comment that (1 * K) == (K * 1) 👍